woocommerce/woocommerce-android
# Refactor to split Order filters view model into 2 view models

Work in progress: #4934

### Description
Coming from this [discussion](https://github.com/woocommerce/woocommerce-android/pull/5039#pullrequestreview-791649968), we agreed that splitting the OrderFilterViewModel into 2 different view models.

Relevant changes happen in files: 
- OrderFilterCategoriesViewModel
- OrderFilterOptionsViewModel
- OrderFiltersUseCases

This code changes also fix an important issue: order filters fragment state loss when avtivity re-creation or process death

### Testing instructions
- Go to dev settings and select "Don't keep activities" option
- Open orders list
- Click filters button
- Select some filters
- Open another app and come back to WooCommerce
- Check filter selections are not lost
- Apply filers and check list is refreshed after "Show filters" button is pressed

### Images/gif


https://user-images.githubusercontent.com/2663464/139917710-aa5b702b-fb4c-4933-9bf4-639673fe9086.mp4
